product_internal_reference_code/models/product_product.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2017 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    @api.model
+    def create(self, vals):
+        if not vals.get("default_code", False):
+            # Get the sequence with user company_id
+            seq_code = (
+                self.env["ir.sequence"]
+                .with_context(force_company=self.env.user.company_id.id)
```

Comment:
```suggestion
                .with_company(self.env.company)
```

From https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0

> Replace appearances of .with_context(force_company=...) with .with_company(...).

From https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0

> If you have .env.user.company_id, you need to change it to .env.company if the intended use is to get "current active company".